src/OpenTelemetry.AutoInstrumentation/Configurations/FileBasedConfiguration/SetDocumentConfiguration.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using OpenTelemetry.AutoInstrumentation.Configurations.FileBasedConfiguration.Parser;
+using Vendors.YamlDotNet.Serialization;
+
+namespace OpenTelemetry.AutoInstrumentation.Configurations.FileBasedConfiguration;
+
+[EmptyObjectOnEmptyYaml]
+internal class SetDocumentConfiguration
```

Comment:
It seems strange to me to name the configuration based off one capability. If possible, I'd prefer to name it GraphQL configuration in case we ever need to expand the configuration options